src/Dafda/Producing/Producer.cs

```diff
@@ -53,5 +51,27 @@ public async Task Produce(object message, Dictionary<string, object> headers)
             var dict = headers.ToDictionary( pair => pair.Key, pair => pair.Value.ToString());
             await Produce(message, new Metadata( dict ));
         }
+
+        /// <summary>
+        /// Produce a <paramref name="message"/> on Kafka
+        /// </summary>
+        /// <param name="message">The message</param>
+        /// <param name="context">Context from the consumer</param>
+        public async Task Produce(object message, MessageHandlerContext context)
+        {
+            await Produce(message, context, new Dictionary<string, string>());
+        }
+
+        /// <summary>
+        /// Produce a <paramref name="message"/> on Kafka
+        /// </summary>
+        /// <param name="message">The message</param>
+        /// <param name="context">Context from the consumer</param>
+        /// <param name="headers">The message headers</param>
+        public async Task Produce(object message, MessageHandlerContext context, Dictionary<string, string> headers)

```

Comment:
Hmm the more we expose this as `Dictionary<string, string>` the less opinionated we can (safely) be about the content within - we already have a bit of logic around this with some `string.IsNullOrEmpty` checks, which could be safer to expose to clients through a type where an empty string is not a representable value. Would it be reasonable to expose this as just the `Metadata` type?

Reply:
I do agree. However the problem here is that the context is filling out the CorrelationId and the CausationId. By adding the Metadata object, we have a conflict with which object to take those named fields from.
I just belive that when implemented this way, it is much easier understandable where data for the named fields are coming from :)